elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metadata/MetaDataStore.java

```diff
@@ -158,35 +192,39 @@ private void addColumn(Column column) {
      *
      * @param metricFunction metric function metadata
      */
-    private void addMetricFunction(MetricFunction metricFunction) {
-        addMetaData(metricFunction);
-        metricFunction.getArguments().forEach(this::addFunctionArgument);
+    private void addMetricFunction(MetricFunction metricFunction, String version) {
+        addMetaData(metricFunction, version);
+        metricFunction.getArguments().forEach(arg -> addFunctionArgument(arg, version));
     }
 
     /**
      * Add a function argument metadata object.
      *
      * @param functionArgument function argument metadata
      */
-    private void addFunctionArgument(FunctionArgument functionArgument) {
-        addMetaData(functionArgument);
+    private void addFunctionArgument(FunctionArgument functionArgument, String version) {
+        addMetaData(functionArgument, version);
     }
 
     /**
      * Add a time dimension grain metadata object.
      *
      * @param timeDimensionGrain time dimension grain metadata
      */
-    private void addTimeDimensionGrain(TimeDimensionGrain timeDimensionGrain) {
-        addMetaData(timeDimensionGrain);
+    private void addTimeDimensionGrain(TimeDimensionGrain timeDimensionGrain, String version) {
+        addMetaData(timeDimensionGrain, version);
     }
 
     /**
      * Add a meta data object into this data store, check for duplication.
      *
      * @param object a meta data object
      */
-    private void addMetaData(Object object) {
+    private void addMetaData(Object object, String version) {
+
+        HashMapDataStore hashMapDataStore = hashMapDataStores.computeIfAbsent(version, ERROR_OUT);
+        EntityDictionary dictionary = hashMapDataStore.getDictionary();
+        Map<Class<?>, Map<String, Object>> dataStore = hashMapDataStore.getStorage();
```

Comment:
Instead of exposing the data storage, I wonder if it would be cleaner to add a method to add something to the storage.

Reply:
updated.